common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java

```diff
@@ -162,6 +162,11 @@ public long toEpochMilli() {
     return localDateTime.toInstant(ZoneOffset.UTC).toEpochMilli();
   }
 
+  public long toEpochMicro() {
+    return localDateTime.toEpochSecond(ZoneOffset.UTC) * 1_000_000
```

Comment:
Can we use similar semantics as `toEpochMilli()`?

`return localDateTime.toInstant(ZoneOffset.UTC).toEpochMilli() * 1000 + localDateTime.getNano() / 1000;`

Reply:
Doing so might lead to loss of precision due to integer overflow. Hence we are parsing the value to seconds here.

Reply:
`localDateTime.toInstant(ZoneOffset.UTC).toEpochMilli()` returns a long value and due to addition of all these values the implicit datatype is long hence there should be no chance of integer overflow.

Reply:
I had tried this and it resulted to wrong results. 
<img width="3374" height="1940" alt="image" src="https://github.com/user-attachments/assets/9d7b57b8-f33f-408c-8f22-569ae2c6e9ee" />

Reply:
Tests are failing with the mentioned code:
<img width="3374" height="1940" alt="image" src="https://github.com/user-attachments/assets/0b6c9136-f0d4-4c61-8238-c5744d01f096" />